bot/exts/moderation/incidents.py

```diff
@@ -394,18 +484,76 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
                 return
 
             if not is_incident(message):
-                log.debug("Ignoring event for a non-incident message")
+                log.debug("Ignoring event for a non-incident message.")
                 return
 
             await self.process_event(str(payload.emoji), message, payload.member)
-            log.trace("Releasing event lock")
+            log.trace("Releasing event lock.")
 
     @Cog.listener()
     async def on_message(self, message: discord.Message) -> None:
         """Pass `message` to `add_signals` if and only if it satisfies `is_incident`."""
         if is_incident(message):
+            message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+            if message_links:
+                embeds = []
+                for message_link in message_links:
+                    ctx = await self.bot.get_context(message)
+                    embeds.append(await make_message_link_embed(ctx, message_link[0]))
+
+                webhook = await self.bot.fetch_webhook(Webhooks.incidents)
+                webhook_embed_list = list(grouper(embeds, 10))
+
+                await self.send_webhooks(webhook_embed_list, message, webhook)
+
+            log.trace(
+                f"Skipping discord message link detection on {message.id}: message doesn't qualify."
+            )
```

Comment:
I'm wondering if this part should be exported to a helper function, and called again on message edits.

Reply:
Meaning, the embeds of the message should be deleted and resent

Reply:
Or just edit the embeds message? unsure

Reply:
Well we have two cases here:

1. If we support more than 10 links, then say we get 10 links in the message, and the author adds another, it would be 11 links, message edit is not possible in this case. In this case I would go with message delete and send new webhook message.

2. If we don't support more than 10 links, then we can just edit the embeds message.

Reply:
Since it seems like we're not supporting more than 10, let's make it work for edits.

Reply:
Resolved in commit 93f9243.